base/docs/helpdb.jl

```diff
@@ -14567,6 +14567,21 @@ include
 doc"""
 ```rst
 ::
+           include_dependency(path::AbstractString)
+
+In a module, declare that the file specified by `path` (relative or
+absolute) is a dependency for precompilation; that is, the
+module will need to be recompiled if this file changes.
+
+This is only needed if your module depends on a file that is not
+used via `include`.   It has no effect outside of compilation.
+```
+"""
+include_dependency
```

Comment:
Putting the docs inline should work fine. As with the rest of the docstrings, `genstdlib.jl` will look for `.. function:: include_dependency(...)` and splice the doc string it finds there, so if that doesn't exist already it needs to be added (or there wouldn't be any way to know where the doc string should go).

When I have more free time I'm going to put up a bunch of example of how to work with this, which will hopefully make it clearer.